crates/sui-adapter-transactional-tests/tests/shared/by_value_shared_object.move

```diff
@@ -25,6 +30,26 @@ module t2::o2 {
         let Obj2 { id } = o2;
         object::delete(id);
     }
+
+    public entry fun freeze_o2(o2: Obj2) {
+        transfer::freeze_object(o2);
+    }
+
+    public entry fun transfer_to_single_owner(o2: Obj2, ctx: &mut TxContext) {
+        transfer::transfer(o2, tx_context::sender(ctx))
+    }
+
+    public entry fun re_share_o2(o2: Obj2) {
+        transfer::public_share_object(o2)
+    }
+
+    public entry fun re_share_non_public_o2(o2: Obj2) {
+        transfer::share_object(o2)
+    }
+
+    public entry fun wrap_o2(o2: Obj2, ctx: &mut TxContext) {
```

Comment:
Is there a test for making a shared object a dynamic object field of another object? If not, we should add one.